src/ChosenSubjects.js

```diff
@@ -1,24 +1,113 @@
 import React from 'react';
 import {makeStyles} from "@material-ui/core/styles";
 import {useSubjects} from "./hooks";
+import DeleteIcon from '@material-ui/icons/Delete';
+import { Button } from '@material-ui/core';
+import {DataGrid} from "@material-ui/data-grid";
+import {subjectColumns} from "./ChosenSubjectsColumns"
 
 const useStyles = makeStyles((theme) => ({
     root: {
         margin: "100px"
     }
 }));
 
+// vychozi razeni
+const sortModel = [
+    {
+        field: 'code',
+        sort: 'asc',
+    },
+];
+
+const getSubjectRows = subjectsFromJSON => {
+    // priprav predmety z JSONu pro zobrazeni v tabulce
+    // pro zobrazeni v tabulce je nutne, aby kazdy zaznam mel ciselne ID, to pak urcuje default sort
+    for (var i = 0; i < subjectsFromJSON.length; i++)
+    {
+        subjectsFromJSON[i].id = i;
+    }
+    return subjectsFromJSON;
+}
+
+const getSubjectColumns = subjects => {
+    // priprav sloupce dynamicky z dat
+    var columns = []
+    for (var prop in subjects[0])
+    {
+        if (prop != 'id')
+            columns.push(prop);
+    }
+    // nejak to serad
+    columns.sort();
+    // takhle to DataGrid vyzaduje - sloupce tabulky jsou samostatne objekty
+    columns = columns.map(x => {return {field: x.toString()}});
+    return columns;
+}
+
+
+const updateTotalRow = subjects => {
```

Comment:
Na toto se vztahují všechny připomínky týkající se js jako takového. Jinak určitě bych přidal nějaký takovýhle součet atd, ale udělal bych to v rámci iterativního přístupu v další iteraci. Musím přiznat, že se mi nelíbí jak to teď je v separátní data grid. To se na to nehodí a určitě bude lepší způsob. Zatím bych tuhle funkci smazal (neboj zůstane na githubu) a v další iteraci ji znovuzrodil.